Userland/Libraries/LibELF/DynamicObject.cpp

```diff
@@ -216,10 +216,39 @@ void DynamicObject::parse()
         m_size_of_relocation_table -= m_size_of_plt_relocation_entry_list;
     }
 
-    auto hash_section_address = hash_section().address().as_ptr();
-    // TODO: consider base address - it might not be zero
-    auto num_hash_chains = ((u32*)hash_section_address)[1];
-    m_symbol_count = num_hash_chains;
+    u32 const* hash_table_begin = (u32*)hash_section().address().as_ptr();
+
+    if (m_hash_type == HashType::SYSV) {
+        u32 n_chain = hash_table_begin[1];
+        m_symbol_count = n_chain;
+    } else {
```

Comment:
As far as I can tell you can make this an early return instead and avoid needing to wrap the rest in one giant else block, to decrease the rightward drift